### ISSUE TITLE
`enum_glob_use` suggestion doesn't match docs

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16900]
### Summary

The suggestion emitted by the `enum_glob_use` lint does not align with its documented guidance. The lint recommends importing enum variants into scope, whereas the documentation indicates that using fully qualified (prefixed) variant names is the preferred approach.

Context: https://github.com/pendulum-project/ntpd-rs/commit/48ea93ee39258c78310f9edcae2ef11f646322fe
Reproducer: https://rust.godbolt.org/z/9618T4s9v

### Reproducer

I tried this code:

```rust
pub enum Foo {
    A,
    B,
}

use Foo::*;

pub fn main() {
    let _ = A;
}
```

I expected to see this happen:

```rust
pub fn main() {
    let _ = Foo::A;
}
```

Instead, this happened:

```rust
use Foo::A;

pub fn main() {
    let _ = A;
}
```

### Version

```text
nightly
```

### Additional Labels

@rustbot label +L-suggestion

**Comment by QEEK-AI** [comment]
**QEEK.AI Agent:** 

"Thanks for reporting this inconsistency! You're absolutely right , there's a clear mismatch between the lint's documented behavior and the actual suggestion it emits."

**Current behavior:**
- **Suggestion:** `use Foo::A;` (import specific variant)
- **Documentation:** Recommends using fully qualified names (`Foo::A`)

**The conflict:**
The documentation likely advocates for fully qualified names to avoid namespace pollution and make code origins explicit. However, the autofix currently takes a more "minimal change" approach by just expanding the glob import to explicit imports.

**Questions to resolve this:**

1. **What should the intended behavior be?**
   - Option A: Fix the suggestion to recommend using `Foo::A` directly (matches current docs)
   - Option B: Update the documentation to match the current suggestion behavior

2. **If we go with Option A:** Should the lint provide both suggestions?
   - Suggest using `Foo::A` as the primary fix
   - Offer `use Foo::A;` as an alternative for cases where the variant is used frequently?

3. **For the referenced PR (pendulum-project/ntpd-rs#48ea93e):** The lint was triggered but the suggestion didn't align with the codebase's preferred style. Did you end up manually changing to `Foo::A` or did you use the `use Foo::A;` form?

My personal preference would be **Option A** , the lint should suggest using fully qualified names since that better aligns with the lint's purpose (reducing glob use and improving code clarity). The `use Foo::A;` form still has the same scoping concerns that `use Foo::*;` has.

Would love to hear thoughts from @rust-lang/clippy on the intended direction here.

**Comment by Souradip121** [comment]
@rustbot claim

Confirmed the mismatch. The suggestion code in wildcard_imports.rs expands use Enum::* into use Enum::Variant, but the docs Use instead example shows importing the enum and writing Enum::Variant at call sites.

Two ways to resolve this. Updating the docs to match the current behavior is a one line change. Changing the suggestion to match the docs is harder, it needs a multispan rewrite of every variant use site and applicability probably drops to MaybeIncorrect.

Could a clippy team member confirm the preferred direction before I open a PR? My read is that updating the docs is safer since the current behavior has been stable since pre 1.29.0 and existing users likely rely on cargo fix producing it.
